posthog/temporal/batch_exports/bigquery_batch_export.py

```diff
@@ -295,9 +295,14 @@ async def acheck_for_query_permissions_on_table(
     ):
         """Attempt to SELECT from table to check for query permissions."""
         job_config = bigquery.QueryJobConfig()
-        query = f"""
-        SELECT 1 FROM  `{table.full_table_id.replace(":", ".", 1)}`
-        """
+        if "timestamp" in [field.name for field in table.schema]:
+            query = f"""
+            SELECT 1 FROM  `{table.full_table_id.replace(":", ".", 1)}` WHERE timestamp IS NOT NULL
```

Comment:
Whether the `WHERE` clause is actually doing something doesn't matter. BigQuery just checks a filter is there.